### PR TITLE
Upgraded firebase_core to 0.5.0

### LINKED ISSUE #8: Compatibility with cloud_firestore@0.14.0
Not sure if it's the right time to upgrade, but 0.14.0 exists now, so I guess at some point it will be.

### PR DESCRIPTION
**Pull request by CiriousJoker**
Breaking changes:
- User -> LitUser (freezed doesn't support prefixes)
- FirebaseUser -> User
- getSignedInUser() is no longer async

Fixes #8

### CODE REVIEW

**Comment by CiriousJoker** [comment]
Fixes #8

File: lib/src/presentation/lit_auth_ui.dart [hunk 1]
```diff
@@ -46,9 +46,9 @@ class LitAuthState extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final user = context.watch<User>();
+    final user = context.watch<LitUser>();
     return user.when(
-      (user) => authenticated,
+      (dynamic user) => authenticated,
```
Comment (funwithflutter) [inline lib/src/presentation/lit_auth_ui.dart:51]: Scratch the dynamic here

**Review by funwithflutter — CHANGES_REQUESTED** [review]
Looks good! Thanks. I'll need to update the docs and the example to add `Firebase.initializeApp()`. Once this is merged I'll do that update the Pub version!

File: lib/src/presentation/lit_auth_ui.dart (hunk 1 above)
Comment (CiriousJoker) [reply lib/src/presentation/lit_auth_ui.dart:51]: Without the dynamics, the compiler complained

**Pull request merged**